### PR TITLE
Add Ally Invest + Merge Portfolios

### PR DESCRIPTION
**Pull request by jmaslek**
In this PR, I added another broker (Ally Invest) and the ability to view holdings across all logged in brokers.

At the time, the Ally Invest does not offer historical portfolio data, so I have not made a "Merged: hist" option.

### CODE REVIEW

File: notebooks/templates/due_diligence.ipynb [hunk 1]
```diff
@@ -185,13 +185,37 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 1,
```
Comment (aia) [inline notebooks/templates/due_diligence.ipynb:188]: I think you accidentally committed an update to the notebook template.

**Comment by aia** [comment]
For portfolio analysis we will want to come up with some kind of a store.

We store lot information in a table. Something like <account> <asset> <quantity> <start date> <end date>.

Based on lot table we can calculate the typical portfolio metrics like returns, volatility, ratios, drawdown.

File: notebooks/templates/due_diligence.ipynb (hunk 1 above)
Comment (jmaslek) [reply notebooks/templates/due_diligence.ipynb:188]: Yes I see - I ran a cell to check the mpg package.  I can't figure out how to reset it.

**Comment by jmaslek** [comment]
So I am not sure I completely follow.  Are you thinking some kind of (ie) SQL table that has all that information (i.e pull account history including sales and whatnot?) 

Obviously holdings doesn't capture your previous returns if you sell, so that is not enough information to do calculations (aside from forecasting or optimizing).  But I would think that the historical portfolios I pull from the 2 brokers would be able to find returns/draws/etc.

**Comment by aia** [comment]
> So I am not sure I completely follow. Are you thinking some kind of (ie) SQL table that has all that information (i.e pull account history including sales and whatnot?)

For now I'd pick something that integrates with Pandas easily - Pickle (I use it in https://github.com/DidierRLopes/GamestonkTerminal/blob/main/notebooks/templates/due_diligence.ipynb), CSV or Parquet. I plan on exploring meta-dbs like LevelDB, SQLlite for storing ticker data.

> Obviously holdings doesn't capture your previous returns if you sell, so that is not enough information to do calculations (aside from forecasting or optimizing). But I would think that the historical portfolios I pull from the 2 brokers would be able to find returns/draws/etc.

I'd just pull lots and calculate portfolio metrics ourselves. Information and metrics provided by brokers are inconsistent and brokers often mess up the cost basis of your lots or even which lots you sold. Lot selection has massive tax implications.

File: notebooks/templates/due_diligence.ipynb (hunk 1 above)
Comment (aia) [reply notebooks/templates/due_diligence.ipynb]: You want to copy the file from the repo, overwrite the copy that you have on the branch and commit that.

Comment (jmaslek) [reply notebooks/templates/due_diligence.ipynb]: I also found that doing git `checkout [last commit on main branch] -- path_to_file` worked

**Comment by jmaslek** [comment]
> For now I'd pick something that integrates with Pandas easily - Pickle (I use it in https://github.com/DidierRLopes/GamestonkTerminal/blob/main/notebooks/templates/due_diligence.ipynb), CSV or Parquet. I plan on exploring meta-dbs like LevelDB, SQLlite for storing ticker data.
> 

Definitely doable -at least the brokers I have now return historical buy/sell information in either csv or a pandas df, so creating a total orders table will be easy.

> I'd just pull lots and calculate portfolio metrics ourselves. Information and metrics provided by brokers are inconsistent and brokers often mess up the cost basis of your lots or even which lots you sold. Lot selection has massive tax implications.

Makes sense.  This is definitely ahead in my roadmap.  We can probably discuss this offline - especially since I am not seeing the easiest route accomplish this -- I guess it would be [something along the lines of this](https://towardsdatascience.com/modeling-your-stock-portfolio-performance-with-python-fbba4ef2ef11) :

- pull all orders
- pull historical price data for each stock 
- calculate a daily portfolio value (price*quantity for each stock held at the time) - probably just based on close price
- analysis

**Review by aia — APPROVED** [review]

**Comment by aia** [comment]
https://towardsdatascience.com/modeling-your-stock-portfolio-performance-with-python-fbba4ef2ef11 - that's not a very optimal approach

You have build the following tables:

Lots:
`<lot-1> <account> <asset> <quantity> <start date> <end date>`

Tickers:
`<time-range (hour, day, month)> <price for ticker-A> <price for ticker-B> <price for ticker-C>`

and you build a portfolio lot value table:
`<time-range (hour, day, month)> <portfolio value of lot 1> <portfolio value of lot 2> <portfolio value of lot 3> ...`

based on lot value table you can calculate total portfolio for any historical timestamp, you can calculate returns and tax adjusted returns (long-term vs short-term vs dividend) and you can calculate the rest of the metrics like volatility, ratios, drawdown.

**Pull request merged**